### ISSUE TITLE
two_component overwrites each dust law's published defaults: kriek_conroy loses its 2175 A bump (3 of 22 laws, up to 128%)

### ISSUE THREAD

**Issue by suchethac** [suchethac/tengri#1833]
Selecting `law_bc='kriek_conroy'` under `dust={'type': 'two_component'}` — the geometry every shipped recipe uses — returns Kriek & Conroy 2013 **with its 2175 Å bump removed**. The same name under `single_component` returns the published law. Three of 22 laws are affected; the SED differs by up to **128%**.

The two dust geometries are two independent implementations of what is, at `tau_bc = 0`, the same screen. Nineteen laws agree between them to machine precision. Three do not, and the three that do not are exactly the three that carry a non-zero shape default in their own signature.

Measured on `48d4b1efb` (current main), public API only.

## The measurement

Fixture: dpl SFH, `redshift=0.5`, bands 1500–6200 Å, an SSP with a Lyman break. `single_component(tau_v=1.0)` against `two_component(tau_bc=0.0, tau_diff=1.0)` — the configuration in which the two are the same model. Max relative difference on `rest_sed()`:

**Live control**: `calzetti` with `tau_v` 0 → 1 moves the SED by rel **1.0**. Dust is fully live on this fixture, so a null result is a real null.

| law | max rel diff | | law | max rel diff |
|---|---|---|---|---|
| calzetti | 3.73e-16 | | prevot_smc | 5.53e-16 |
| cardelli | 4.28e-16 | | reddy15 | 4.26e-16 |
| conroy2010 | 4.96e-16 | | salim | 4.25e-16 |
| d03_mwrv31 | 4.43e-16 | | salim_sbl18 | 3.56e-16 |
| hd23_mwrv31 | 4.34e-16 | | smc | 4.41e-16 |
| **kriek_conroy** | **1.281372e+00** | | **tea** | **1.000000e+00** |
| leitherer02 | 3.56e-16 | | vw07_bc | 4.20e-16 |
| li08 | 4.39e-16 | | vw07_diff | 4.28e-16 |
| lmc | 4.52e-16 | | wd01_mwrv31 | 4.29e-16 |
| **narayanan_z** | **1.000000e+00** | | wd01_smcbar | 4.39e-16 |
| noll09 | 3.56e-16 | | power_law | 4.28e-16 |

Nineteen inert results beside three live ones. By the repo's own discriminator that indicts the scope, not the fixture.

## Which path is wrong

Each of the three carries its paper's value as a signature default, while the spec declares one shared `dust_delta` / `dust_bump_strength`, both `Fixed(0.0)`:

```
kriek_conroy    {'dust_bump_strength': 1.0, 'dust_delta': 0.0}
narayanan_z     {'dust_delta': -0.2, 'dust_bump_strength': 1.0, 'redshift': 0.0}
tea             {'dust_delta': -0.2, 'dust_tea_scatter': 0.0}
```

Forcing the shared knob to an explicit value on both paths isolates the direction. For `kriek_conroy`:

| build | single-vs-two | single moved | two moved |
|---|---|---|---|
| defaults | **1.281e+00** | — | — |
| `bump_strength=1.0` | 4.19e-16 | **0.000e+00** | **1.281e+00** |
| `bump_strength=0.0` | 4.25e-16 | 5.617e-01 | 0.000e+00 |

`single_component` does not move when the paper value is stated explicitly, because it was already using it. `two_component` moves by the full discrepancy, because it was using `0.0`. Same direction for `narayanan_z` and `tea` under `delta=-0.2` (single moves 0.0, two moves 1.0, and they then agree to 4.3e-16).

**`single_component` honors each law's published defaults. `two_component` overwrites them with the spec's `Fixed(0.0)`.**

For `kriek_conroy` that is not a subtle shift. `dust_bump_strength` is documented as a multiplier on the KC13 Drude amplitude where "0 removes the bump" (`attenuation.py:378-382`). A user who selects Kriek & Conroy gets Calzetti-with-a-tilt.

## Why this reaches users

All ten shipped recipes build `two_component`; **none** build `single_component` (`recipes/__init__.py`: 90, 145, 209, 270, 326, 399, 468, 518, 583, 636). The model-construction examples in `CLAUDE.md` and `docs/dev/model-construction.md` use `two_component` throughout. The defective path is the recommended one, and the correct path is the one nothing steers you to.

## Root cause: one model, two implementations

`DustAttenuationSEDComponent` (`components/dust/component.py`, 413 lines) and `DustSEDComponent` (`components/dust/two_component.py`, 876 lines) implement the same screen. Sixteen call sites across four modules resolve a dust law and evaluate a curve — `attenuation.py` ×3, `two_component.py` ×6, `_apply.py` ×5, `component.py` ×2 — so a change to how a law's arguments are bound has sixteen places to be right and no single place that is authoritative.

The stated reason for the split is satisfied and stale. `component.py:117-119`:

> Two-component (Charlot & Fall 2000) attenuation will be a separate adapter **once the stellar component publishes per-age luminosities**.

Stellar publishes `lnu_age`; `two_component.py:493` raises if it is absent. The condition was met and the split was never revisited.

The performance argument is the only live one: `two_component` needs the per-age cube because the birth-cloud screen is age-dependent, whereas `single_component` reads only `sed_intrinsic`, so its screen factors out of the age integral. That is a real saving and an argument for keeping one fast path — not for keeping two independent bindings of the law arguments.

## What this is *not*

Worth recording, because it is the first thing a reader will suspect and it is false.

`two_component.py:713-717` guards the non-stellar SED explicitly — *"Stellar dust does not attenuate AGN/nebular/radio/xray"* — while `single_component` writes `sed_intrinsic * exp(-tau*k)` over everything upstream. That looks like a physics divergence. It is unreachable: AGN, radio and X-ray all run **downstream** of dust in both chains.

```
single -> ['stellar', 'nebular', 'dust_attenuation', 'agn', 'radio', 'xray', 'igm']
two    -> ['stellar', 'nebular', 'dust',             'agn', 'radio', 'xray', 'igm']
```

Measured with an AGN at **99.5%** of the MIR–FIR (`agn_log_lbol` swept to the top of its declared support, stellar scaled down so the AGN is a live contributor): single-vs-two **exactly 0.0** there, 4.2e-16 in the NIR at 55.7% AGN share. The subtraction only ever isolates the nebular continuum, which `two_component` re-adds attenuated by the birth-cloud law — and at `tau_bc=0` that is the identity. The guard is defensive against a case the topological sort prevents.

## History

This is the defect #1808 declined to introduce, sitting on the other path the whole time. That PR implemented "pass the spec's values unconditionally", measured it, and rejected it because it "collapsed three distinct published laws onto one curve". The three laws were `kriek_conroy`, `narayanan_z` and `tea`. `two_component` has always done exactly that, and #1808 fixed only `single_component`.

#844 folded both onto `SEDModelComponent` with the checklist item *"DustAttenuationSEDComponent / two_component as SEDModelComponent (bit-exact)"*. Bit-exactness was verified for each component against its own previous behaviour, never for the two against each other — which is the comparison that fails.

Related but distinct: #1731 (k(5500 Å)=1 normalization honored by some laws and not others, ≤0.29%) is about the laws; this is about how two callers bind their arguments.

## Proposed fix

**1. Repair `two_component` (small, and it is a correctness bug on the recommended path).** Apply the provenance rule #1808 established for `single_component`: `user_prior` / `user_fixed` / `user_free` / `wildcard_free` are requests and get passed; `registry_default` / `wildcard_fixed` are not, and the law's own default stands. The resolution already exists in `_build_component_chain`; `two_component` needs to consult it. Regression test: for every law in the registry, `single_component(tau_v=t)` and `two_component(tau_bc=0, tau_diff=t)` must agree to `_INERT_TOL`, with the calzetti sweep as the live control — i.e. the table above, as an assertion.

**2. Then consolidate.** One component with an age-dependent screen and a fast path when `tau_bc` is fixed at 0, or one shared law-binding seam both components call. The second is smaller and captures most of the value: the bug exists because sixteen sites bind law arguments and only one of them learned the rule.

Doing (1) without (2) leaves the next law with a non-zero default to be got wrong again by whichever site is not updated. Doing (2) without (1) risks preserving the wrong behaviour as the consolidated one — `two_component` is the path with the users, so a naive merge would keep its binding.

## Reproduction

```python
import warnings; warnings.simplefilter("ignore")
import jax, jax.numpy as jnp, numpy as np
from tengri import FIXED, Fixed, Observation, Photometry, SEDModel, SSPData
from tengri.observation.photometry import FilterCurve

ages, lgmet = jnp.linspace(-3.0, 1.14, 25), jnp.array([-4.0, -2.65, -1.3])
wave = jnp.logspace(2.0, 7.0, 1200)
base = ((5000.0 / wave) ** 2 * jnp.where(wave < 912.0, 1e-6, 1.0))[None, None, :]
flux = (base * (1.0 + 0.15 * (ages - ages.mean()))[None, :, None]
             * (1.0 + 0.10 * (lgmet - lgmet.mean()))[:, None, None])
ssp = SSPData(ssp_wave=wave, ssp_flux=jnp.abs(flux) + 1e-12, ssp_lg_age_gyr=ages, ssp_lgmet=lgmet)

def band(c, n=32):
    w = jnp.linspace(c * 0.8, c * 1.2, n)
    return FilterCurve(wave=w, trans=jnp.sin(jnp.linspace(0, jnp.pi, n)) * 0.6, name=f"b{c:.4g}")
obs = Observation(photometry=Photometry(filters=tuple(band(c) for c in (1500., 2800., 3500., 6200.))))

def sed(dust):
    m = SEDModel.build(ssp_data=ssp, observation=obs, sfh={"type": "dpl", "all_params": FIXED},
                       dust=dust, redshift=Fixed(0.5))
    return np.asarray(m.predict(dict(m.spec.sample(jax.random.PRNGKey(0)))).rest_sed())

LAW = "kriek_conroy"
a = sed({"type": "single_component", "law_bc": LAW, "tau_v": Fixed(1.0)})
b = sed({"type": "two_component", "law_bc": LAW, "law_diff": LAW,
         "tau_bc": Fixed(0.0), "tau_diff": Fixed(1.0)})
rel = np.abs(a - b) / np.where(np.abs(b) > 0, np.abs(b), 1.0)
print("max rel:", rel.max())   # 1.281372e+00   (calzetti in the same harness: 3.73e-16)
```
